feat: show git commit hash in version

michael1011 opened the pull request:
Closes #1171 

I created a new script (`parseGitCommit.js`) that will be executed before the `compile` NPM script. It parses the latest git commit hash (if that commit doesn't also tag a release) and whether there are uncommitted changes in the working directory and writes a corresponding string into `lib/Version.ts`.

For example the `version` in `xucli getinfo -j` could look like: `1.0.0-testnet.1-9a03442c-dirty`

kilrau commented on parseGitCommit.js:
```diff
@@ -0,0 +1,36 @@
+  }
+};
+
+const isDirty = () => {
```
Nice!

kilrau commented on lib/service/Service.ts:
```diff
@@ -269,7 +270,7 @@ class Service {
       uris,
       numPairs,
-      version: this.version,
+      version: `${this.version}${commitHash}`,
```
I would go for `${this.version}${commitHash}`, since imo the version name should only be on the official release.

`version` in `xucli getinfo -j` should look like: `master-9a03442c-dirty`

michael1011 replied:
Are you sure about that? With the current approach if the commit is also the tag it will just show the version from `package.json`. If not it will give you the `package.json` version and the commit hash. This is enough information to know whether you are running the release I think.

Also it is really nice to know on what release the branch you are running is based on.

kilrau replied:
> Also it is really nice to know on what release the branch you are running is based on.

Ok, I can see that. Fine then.

michael1011 commented:
I just addressed your comments @sangaman! Thanks for the review!

I also just tried tagging a release and it worked fine for me. Could you give it another try?

sangaman commented on parseGitCommit.js:
```diff
@@ -0,0 +1,36 @@
+
+try {
+  // Delete the version file is it exists
+  if (fs.access(versionFilePath)) {
```
Duh, I forgot about `unlinkSync`. Definitely the simplest way to do it.
